cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/etl/api/batch/BatchSource.java

```diff
@@ -37,7 +38,8 @@
  */
 @Beta
 public abstract class BatchSource<KEY_IN, VAL_IN, OUT> extends BatchConfigurable<BatchSourceContext>
-  implements Transformation<KeyValue<KEY_IN, VAL_IN>, OUT>, StageLifecycle<BatchRuntimeContext> {
+  implements Transformation<KeyValue<KEY_IN, VAL_IN>, OUT>, StageLifecycle<BatchRuntimeContext> { // ,
+//  SubmitterLifecycle<BatchSourceContext> {
```

Comment:
if we don't have StageSubmitter, BatchConfigurable can be changed to implement SubmitterLifecycle and you can remove this comment